PR: https://github.com/odoo/odoo/pull/220937

From: a245bfa061ed43e5b51329a0722311fa6fe5ab5f
From: hapt-odoo
Date: 2025-11-21 17:49:29

Breaking data model changes scores: del:1 + add:0, change matches:
-    report_grids = fields.Boolean(string="Print Variant Grids", default=True, help="If set, the matrix of configurable products will be shown on the report of this order.")

Total Changes: 27

[IMP] purchase: enhance UX of 'Other Info' tab layout in PO

* Renamed the 'Other Information' tab to 'Other Info' for better readability
  and simplified navigation.
* Introduced three sections within the 'Other Info' tab — Purchase, Billing
  and Tracking — to better organize fields based on their purpose and
  improve the overall layout. which made easy to deal with all the fields.
* Rearranged fields in the tab according to their functional relevance and
  moved them to appropriate  sections to enhance user experience.
* Payment Terms field has been moved to the main form view for better
  visibility and quicker access.
* Removed the 'Print Variant Grid' boolean from the view. This option was
  previously invisible in debug mode. Users who do not require the grid can
  now manage this via the report, ensuring cleaner code and simplified user
  control.
* These changes collectively deliver a clearer and more simplified view for
  users, improving efficiency and usability.

Task Id: 4778630

closes odoo/odoo#220937

Related: odoo/enterprise#91195
Related: odoo/upgrade#8312
Signed-off-by: Arnold Moyaux (arm) <arm@odoo.com>

================================= pseudo patch: =================================

--- a/addons/purchase_product_matrix/models/purchase.py
+++ b/addons/purchase_product_matrix/models/purchase.py
@@ -8,8 +8,6 @@ from odoo.exceptions import ValidationError
 class PurchaseOrder(models.Model):
     _inherit = 'purchase.order'
 
-    report_grids = fields.Boolean(string="Print Variant Grids", default=True, help="If set, the matrix of configurable products will be shown on the report of this order.")
-
     """ Matrix loading and update: fields and methods :
 
     NOTE: The matrix functionality was done in python, server side, to avoid js
@@ -141,19 +139,18 @@ class PurchaseOrder(models.Model):
     def get_report_matrixes(self):
         """Reporting method."""
         matrixes = []
-        if self.report_grids:
-            grid_configured_templates = self.order_line.filtered('is_configurable_product').product_template_id
-            # TODO is configurable product and product_variant_count > 1
-            # configurable products are only configured through the matrix in purchase, so no need to check product_add_mode.
-            for template in grid_configured_templates:
-                if len(self.order_line.filtered(lambda line: line.product_template_id == template)) > 1:
-                    matrix = self._get_matrix(template)
-                    matrix_data = []
-                    for row in matrix['matrix']:
-                        if any(column['qty'] != 0 for column in row[1:]):
-                            matrix_data.append(row)
-                    matrix['matrix'] = matrix_data
-                    matrixes.append(matrix)
+        grid_configured_templates = self.order_line.filtered('is_configurable_product').product_template_id
+        # TODO is configurable product and product_variant_count > 1
+        # configurable products are only configured through the matrix in purchase, so no need to check product_add_mode.
+        for template in grid_configured_templates:
+            if len(self.order_line.filtered(lambda line: line.product_template_id == template)) > 1:
+                matrix = self._get_matrix(template)
+                matrix_data = []
+                for row in matrix['matrix']:
+                    if any(column['qty'] != 0 for column in row[1:]):
+                        matrix_data.append(row)
+                matrix['matrix'] = matrix_data
+                matrixes.append(matrix)
         return matrixes
 
 
